addons/stock/report/report_lot_barcode.xml

```diff
@@ -3,7 +3,7 @@
 <data>
 <template id="report_lot_label">
     <t t-call="web.basic_layout">
-        <t t-set='nRows' t-value='12'/>
+        <t t-set='nRows' t-value='7'/>
```

Comment:
Following my previous comment, we can change the size of the label itself to match the 4x12 format and change the label's size at the line 22:
```xml
<div t-att-style="'position:relative; width:43mm; height:19mm; border: 1px solid %s;' % (o.env.user.company_id.primary_color or 'black')">
```
I suggest you to use a height of 19mm ([like it was before](https://github.com/odoo/odoo/commit/2185c7c7303c2738c63ef454b061f85e009d29ee#diff-fbc7f34c633189f45e17080184e52bb403bf631e36d8713cb0b2d27260629a3bL21)) and I think the font size used for the product's name and the lot should be decreased accordingly